internal/runner/rmq.go

```diff
@@ -489,30 +504,65 @@ func (rmq *RabbitMQ) QueueDeclare(qName string) (err kv.Error) {
 		conn.Close()
 	}()
 
-	_, errGo := ch.QueueDeclare(
-		qName, // name
-		false, // durable
-		false, // delete when unused
-		false, // exclusive
-		false, // no-wait
-		nil,   // arguments
-	)
-	if errGo != nil {
-		return kv.Wrap(errGo).With("stack", stack.Trace().TrimRuntime()).With("qName", qName).With("uri", rmq.mgmt).With("exchange", rmq.exchange)
+	if errGo := ch.QueueBind(qName, "StudioML."+qName, "StudioML.topic", false, nil); errGo != nil {
+		return kv.Wrap(errGo).With("stack", stack.Trace().TrimRuntime()).With("qName", qName, "uri", rmq.mgmt, "exchange", rmq.exchange)
 	}
 
-	if errGo = ch.QueueBind(qName, "StudioML."+qName, "StudioML.topic", false, nil); errGo != nil {
-		return kv.Wrap(errGo).With("stack", stack.Trace().TrimRuntime()).With("qName", qName).With("uri", rmq.mgmt).With("exchange", rmq.exchange)
+	return nil
+}
+
+func (rmq *RabbitMQ) declareQVariants(name string) (err kv.Error) {
+	// First we try using the exchange in either tranisent or persistent mode if it can be found.  This is done by trying to use the passive mode
+	// to find queues that already exist and if they dont default to persistent mode, this will be the case for testing.
+	//
+	if err = rmq.declareQTranisent(name); err == nil {
+		return nil
 	}
+	// retry on a new connection, rmq has the habbit of killing connections on the first error,
+	// so shadow the variables and add new termination handlers
+	return rmq.declareQPersistent(name)
+}
 
+func (rmq *RabbitMQ) declareQTranisent(name string) (err kv.Error) {
+	conn, ch, err := rmq.attach(rmq.exchange)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		ch.Close()
+		conn.Close()
+	}()
+
+	// Declare parameters are, name, durable, delete when unused,, exclusive, no-wait, arguments
+	if _, errGo := ch.QueueDeclarePassive(name, false, false, false, false, nil); errGo != nil {
+		if errGo != nil {
+			return kv.Wrap(errGo).With("stack", stack.Trace().TrimRuntime()).With("qName", name, "uri", rmq.mgmt, "exchange", rmq.exchange)
+		}
```

Comment:
Why the second check for errGo != nil?

Reply:
Yes, pointless I remember removing something there at somepoint.  Will get rid of the redundancy.